localstack-core/localstack/services/events/target.py

```diff
@@ -434,9 +439,13 @@ def send_event(self, event):
 
 class KinesisTargetSender(TargetSender):
     def send_event(self, event):
-        partition_key_path = self.target["KinesisParameters"]["PartitionKeyPath"]
+        partition_key_path = collections.get_safe(
+            self.target,
+            "$.KinesisParameters.PartitionKeyPath",
+            default_value="$.id",
+        )
         stream_name = self.target["Arn"].split("/")[-1]
-        partition_key = event.get(partition_key_path, event["id"])
+        partition_key = collections.get_safe(event, partition_key_path, event["id"])
```

Comment:
It's fixed :D We should probably add a test in community as well to capture these regressions.